src/main/java/nl/mpcjanssen/simpletask/Simpletask.kt

```diff
@@ -718,26 +718,36 @@ class Simpletask : ThemedNoActionBarActivity() {
     }
 
     private fun deleteTasks(tasks: List<TodoItem>) {
-        showConfirmationDialog(this, R.string.delete_task_message, DialogInterface.OnClickListener { dialogInterface, i ->
+        val numTasks = tasks.size
+        var title = getString(R.string.delete_task_title)
+                    .replaceFirst(Regex("%s"), numTasks.toString())
+        val delete = DialogInterface.OnClickListener { dialogInterface, i ->
             for (t in tasks) {
                 TodoList.remove(t)
             }
             TodoList.notifyChanged(Config.todoFileName, Config.eol, m_app, true)
             invalidateOptionsMenu()
-        }, R.string.delete_task_title)
-    }
+        }
 
-    private fun archiveTasks(tasksToArchive: List<TodoItem>?, areYouSureDialog: Boolean) {
+        showConfirmationDialog(this, R.string.delete_task_message, delete, title)
+    }
 
+    private fun archiveTasks() { archiveTasks(TodoList.completedTasks, false) }
+    private fun archiveTasks(tasks: List<TodoItem>, showDialog: Boolean = true) {
         val archiveAction = {
             if (Config.todoFileName == m_app.doneFileName) {
                 showToastShort(this, "You have the done.txt file opened.")
             }
-            TodoList.archive(Config.todoFileName, m_app.doneFileName, tasksToArchive, Config.eol)
+            TodoList.archive(Config.todoFileName, m_app.doneFileName, tasks, Config.eol)
```

Comment:
Given that we're in the archiveTasks function, I think saving space is more important than redundant descriptive names.